File: backend/SebPortal.Api/Models/Payment.cs
namespace SebPortal.Api.Models;

public class Payment
{
    public int Id { get; set; }
    public int TenantId { get; set; }
    public int FromAccountId { get; set; }
    public string ToIban { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "SEK";
    public string? Reference { get; set; }
    public string Status { get; set; } = PaymentStatuses.PendingApproval;
    public int? CreatedById { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ExecutedAt { get; set; }

    public Account? FromAccount { get; set; }
    public User? CreatedBy { get; set; }
    public Tenant? Tenant { get; set; }
    public ICollection<ApprovalStep> ApprovalSteps { get; set; } = new List<ApprovalStep>();
}

File: frontend/src/types/Payment.ts
export type PaymentStatus = "completed" | "pending_approval";

export type PaymentId = number | string;

export type CreatePaymentRequest = {
  fromAccountId: number;
  toIban: string;
  amount: string;
  reference?: string;
};

export type CreatePaymentResponse = {
  id: PaymentId;
  status: PaymentStatus;
  fromAccountId: number;
  toIban: string;
  amount: string;
  currency: string;
  reference: string;
  createdAt: string;
};
